pkg/devfile/adapters/common/utils.go

```diff
@@ -70,3 +71,34 @@ func GetSupportedComponents(data data.DevfileData) []common.DevfileComponent {
 	}
 	return components
 }
+
+// IsEnvPresent checks if the env variable is present in an array of env variables
+func IsEnvPresent(envVars []common.DockerimageEnv, envVarName string) bool {
+	isPresent := false
+
+	for _, envVar := range envVars {
+		if *envVar.Name == envVarName {
+			isPresent = true
+		}
+	}
+
+	return isPresent
+}
+
+// IsComponentBuildRequired checks if a component build is required based on the push commands, it throws an error
+// if the push commands does not meet the expected criteria
+func IsComponentBuildRequired(pushDevfileCommands []common.DevfileCommand) (bool, error) {
+	var buildRequired bool
+
+	if len(pushDevfileCommands) == 1 {
+		// if there is one command, it is the mandatory run command. No need to build.
+		buildRequired = false
+	} else if len(pushDevfileCommands) == 2 {
+		// if there are two commands, it is the optional build command and the mandatory run command, set buildRequired to true
+		buildRequired = true
+	} else {
+		return false, fmt.Errorf("error executing devfile commands - there should be at least 1 command or at most 2 commands, currently there are %v commands", len(pushDevfileCommands))
```

Comment:
Will this error cause issues when we need to add support devfile init commands (or other commads) later for Docker? Would it be better to just set `buildRequired = true` if `len(pushDevfileCommands) >= 2`?

Reply:
dev init PR changed the logic, which i think is better(glancing at it). That will replace this..